### ISSUE TITLE
Terraform syntax warnings do not check data source or output blocks

### ISSUE THREAD

**Issue by gdavison** [hashicorp/terraform#25828]
The `terraform init` and `terraform validate` commands output syntax warnings for deprecated interpolation syntax for resources, but do not return warnings when the deprecated interpolation is on data source or output blocks.

### Terraform Version

This error occurs in both 0.12 and 0.13

```console
$ terraform version
Terraform v0.12.29
+ provider.aws v3.1.0
```

```console
$ terraform version
Terraform v0.13.0
+ provider registry.terraform.io/hashicorp/aws v3.1.0
```

### Terraform Configuration Files

```terraform
variable "api_gateway_name" {
  type = string
}

data "aws_api_gateway_rest_api" "selected" {
  name = "${var.api_gateway_name}"
}

data "aws_cognito_user_pools" "selected" {
  name = "selected"
}

resource "aws_api_gateway_authorizer" "cognito" {
  name          = "cognito"
  type          = "COGNITO_USER_POOLS"
  rest_api_id   = "${data.aws_api_gateway_rest_api.selected.id}"
  provider_arns = data.aws_cognito_user_pools.selected.arns
}

output "api_id" {
  value = "${aws_api_gateway_authorizer.cognito.rest_api_id}"
}
```

### Expected Behavior

The warning

```
Warning: Interpolation-only expressions are deprecated
```

should be generated for
* data source `aws_api_gateway_rest_api.selected` `name`
* resource `aws_api_gateway_authorizer.cognito` `rest_api_id`
* output `api_id` `value`

### Actual Behavior

The warning

```
Warning: Interpolation-only expressions are deprecated
```

is only generated for
* resource `aws_api_gateway_authorizer.cognito` `rest_api_id`

### Steps to Reproduce

1. `terraform init`
2. (Optionally) `terraform validate`

**Comment by alisdair** [comment]
Thanks for reporting this! [Quoting a previous response from Martin to a related issue](https://github.com/hashicorp/terraform/issues/25430#issuecomment-651398989) (#25430):

> Technically this is working as intended -- it was intentionally only implemented for these two blocks so far -- but it feels weird to mark a deprecation warning as an "enhancement" so I'm going to label this as a bug in the configuration package and we can decide what to do about it at a later time. We generally add new deprecation warnings only later in a major release cycle so folks don't have to change lots of unrelated stuff as part of upgrading to a major release, and 0.13.0 is due in a few weeks so we'll need to wait a little while before potentially adding more deprecation warnings of this type.

There's a pending PR to fix the other issue (#25465), which I think we could reasonably extend to also add these deprecations to data sources and objects.

**Comment by gdavison** [comment]
Thanks, @alisdair! We've been using the syntax warnings (imported by `tflint`) to clean up the AWS Provider documentation and prevent old syntax from getting back in from PRs.
